apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[slug]/(active)/components/client/update-partner-form.client.tsx

```diff
@@ -0,0 +1,249 @@
+"use client";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { cn } from "@/lib/utils";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2 } from "lucide-react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+import { ToolTipLabel } from "../../../../../../../../../@/components/ui/tooltip";
+import type { Ecosystem, Partner } from "../../../../types";
+import { isDomainRegex } from "../../constants";
+import { useUpdatePartner } from "../../hooks/use-update-partner";
+
+const formSchema = z.object({
```

Comment:
can we share all or parts of this form with the "add" form? I assume it's pretty much the same just with optional fields instead of required?

Reply:
then lets :)